alchemiscale/storage/statestore.py

```diff
@@ -876,6 +883,227 @@ def delete_network(
+                # single task.
+                for record in self.task_tree:
+                    etask_record = record["extended_task"]
+                    pdrr_records = record["pdrrs"]
+                    # update task node to have new scoped key
+                    task_node = record_to_node(record["task"])
+                    # create the task node this task extends if it exists
+                    etask_node = (
+                        None
+                        if not record["extended_task"]
+                        else record_to_node(record["extended_task"])
+                    )
+                    if etask_node:
+                        subgraph |= Relationship.type("EXTENDS")(
+                            etask_node, task_node, **scope_props
+                        )
+                    # clone all result refs for the task
+                    for pdrr_record in record["pdrrs"]:
+                        pdrr_node = record_to_node(pdrr_record)
+                        subgraph |= Relationship.type("RESULTS_IN")(
+                            task_node, pdrr_node, **scope_props
+                        )
+
+                return subgraph
+
+        # TODO: upstream to gufe
+        def kc_to_gufe(kc, gts):
+            for gufe_key, keyed_dict in kc:
+                if gt := gts.get(gufe_key):
+                    continue
+                gt = key_decode_dependencies(keyed_dict, registry=gts)
+                gts[gufe_key] = gt
+            return gt
+
+        # Map Transformation and NonTransformation objects to their
+        # potentially duplicated database gufe keys. This may happen
+        # due to minor version changes between serialization
+        # times. These keys are then used to get all results for Tasks
+        # associated with them.
+        transformation_data: list[TransformationData] = []
+        subchain_cache = {}
+        for network_scope, network_keyed_chain in network_keyed_chains:
+            gts = {}
+            for index, (database_key, _) in enumerate(network_keyed_chain):
+                # only process Transformation or NonTransformation keys
+                if (
+                    "Transformation" in database_key
+                    or "NonTransformation" in database_key
+                ):
+                    # get the tokenizable at the index along with all previous data
+                    subchain = KeyedChain(network_keyed_chain[: index + 1])
+                    transformation = kc_to_gufe(subchain, gts)
+                    subchain_cache[transformation] = subchain
+                    try:
+                        idx = transformation_data.index(transformation)
+                        data = transformation_data[idx]
+                    except ValueError:
+                        data = TransformationData(transformation)
+                        transformation_data.append(data)
+                    data.add_known_scoped_key(database_key, network_scope)
```

Comment:
Might be able to now use `GufeTokenizable.decode_subchains` now that OpenFreeEnergy/gufe#629 is closed?